api/src/main/java/org/openmrs/util/OpenmrsUtil.java

```diff
@@ -2149,5 +2152,34 @@ public static boolean isYesterday(Date date) {
 	public static Set<String> getDeclaredFields(Class<?> clazz) {
 		return Arrays.stream(clazz.getDeclaredFields()).map(Field::getName).collect(Collectors.toSet());
 	}
+
+	/**
+	 * Checks if a given value is within the normal reference range for a concept and person.
+	 *
+	 * @param value The value to check
+	 * @param concept The concept associated with the value
+	 * @param person The person associated with the observation
+	 * @return A map where the `key` is a boolean indicating if the value is within range, and the `value` contains a message
+	 * 
+	 * @since 2.7.0
+	 */
+	public static Map<Boolean, String> isInNormalReferenceRange(Float value, Concept concept, Person person) {
```

Comment:
The client does not need a map. Like i said in the original github comments, all they need is call a method like `isValidNumericValue` which either returns a boolean or a string for the error message to display.